transaction/src/plan/memo.rs

```diff
@@ -9,20 +9,30 @@ use rand::{CryptoRng, RngCore};
 
 #[derive(Clone, Debug)]
 pub struct MemoPlan {
-    pub plaintext: MemoPlaintext,
+    pub plaintext: [u8; MEMO_LEN_BYTES],
     pub key: PayloadKey,
 }
 
 impl MemoPlan {
     /// Create a new [`MemoPlan`].
-    pub fn new<R: CryptoRng + RngCore>(rng: &mut R, plaintext: MemoPlaintext) -> MemoPlan {
+    pub fn new<R: CryptoRng + RngCore>(
+        rng: &mut R,
+        plaintext_str: String,
+    ) -> Result<MemoPlan, anyhow::Error> {
+        if plaintext_str.as_bytes().len() > MEMO_LEN_BYTES {
+            return Err(anyhow::anyhow!(
+                "provided plaintext for memoplan exceeds maximum memo size"
+            ));
+        }
         let key = PayloadKey::random_key(rng);
-        MemoPlan { plaintext, key }
+        let mut plaintext = [0u8; MEMO_LEN_BYTES];
+        plaintext[..plaintext_str.as_bytes().len()].copy_from_slice(plaintext_str.as_bytes());
+        Ok(MemoPlan { plaintext, key })
```

Comment:
This logic is replicated from the logic in `MemoCiphertext::encrypt`, which feels like a small code smell. If we do indeed store a `String` in the `MemoPlan` rather than an array of bytes (as I recommend above), then we don't need this check at all, since it will be performed when we try to encrypt the memo in `MemoPlan::memo`.